lib/ex_aws/ec2/impl.ex

```diff
@@ -0,0 +1,1009 @@
+  def reset_snapshot_attribute(client, snapshot_id, attribute, opts \\ []) do
+    query_params = opts
+    |> normalize_opts
+    |> Map.merge(%{
+      "Action"     => "ResetSnapshotAttribute",
+      "Version"    => @version,
+      "SnapshotId" => snapshot_id,
+      "Attribute"  => attribute
+      })
+
+    request(client, :post, "/", params: query_params)
+  end
+
+  ##################################
+  ### Account Attributes Actions ###
+  ##################################
+
+  @params [:dry_run, [:attribute_name]]
+  def describe_account_attributes(client, opts \\ []) do
+    query_params = opts
+    |> normalize_opts
+    |> Map.merge(%{
+      "Action"  => "DescribeAccountAttributes",
+      "Version" => @version
+      })
+
+    request(client, :get, "/", params: query_params)
+  end
+
+  ############################
+  ### Bundle Tasks Actions ###
+  ############################
+
+  @params [:dry_run]
+  def bundle_instance(client, instance_id, {s3_aws_access_key_id, s3_bucket, s3_prefix, s3_upload_policy, s3_upload_policy_sig}, opts \\ []) do
+    query_params = opts
+    |> normalize_opts
+    |> Map.merge(%{
+      "Action"                           => "BundleInstance",
+      "Version"                          => @version,
+      "InstanceId"                       => instance_id,
+      "Storage.S3.AWSAccessKeyId"        => s3_aws_access_key_id,
+      "Storage.S3.Bucket"                => s3_bucket,
+      "Storage.S3.Prefix"                => s3_prefix,
+      "Storage.S3.UploadPolicy"          => s3_upload_policy,
+      "Storage.S3.UploadPolicySignature" => s3_upload_policy_sig
+      })
+
+    request(client, :post, "/", params: query_params)
+  end
+
+  @params [:dry_run]
+  def cancel_bundle_task(client, bundle_id, opts \\ []) do
+    query_params = opts
+    |> normalize_opts
+    |> Map.merge(%{
+      "Action"   => "CancelBundleTask",
+      "Version"  => @version,
+      "BundleId" => bundle_id
+      })
+
+    request(client, :post, "/", params: query_params)
+  end
+
+  @params [[:bundle_id], :dry_run, [:filter]]
+  def describe_bundle_tasks(client, opts \\ []) do
+    query_params = opts
+    |> normalize_opts
+    |> Map.merge(%{
+      "Action"  => "DescribeBundleTasks",
+      "Version" => @version
+      })
+
+    request(client, :get, "/", params: query_params)
+  end
+
+  ########################
+  ### Helper Functions ###
+  ########################
+
+  defp request(%{__struct__: client_module} = client, verb, path, data) do
+    client_module.request(client, verb, path, data)
+  end
+
+  defp normalize_opts(opts) do
+    opts
+    |> Enum.into(%{})
+    |> camelize_keys
+  end 
+
+  defp list_builder([h | []], key, count, state) do
+    Map.put_new(state, "#{key}.#{count}", h)
+  end
+
+  defp list_builder([h | t], key, count, state) do
+    list_builder t, key, count + 1, Map.put_new(state, "#{key}.#{count}", h)
+  end
+
+  defp list_builder_key_val([_h = {f, s} | []], key, count, state) do
```

Comment:
I'm just used to doing it like that for expressiveness and explicit purposes (to make it clearer to someone new to the code that it's the head of the list). I'll remove it.